src/ray/protobuf/core_worker.proto

```diff
@@ -186,23 +186,23 @@ message WaitForActorOutOfScopeReply {
 message UpdateObjectLocationBatchRequest {
   bytes intended_worker_id = 1;
   bytes node_id = 2;
-  repeated ObjectLocationStateUpdate object_location_states = 3;
+  repeated ObjectLocationUpdate object_location_updates = 3;
 }
 
 message UpdateObjectLocationBatchReply {
 }
 
-enum ObjectLocationState {
-  // The latest object state.
-  // Object is added.
-  ADDED = 0;
-  // Object is removed.
-  REMOVED = 1;
-}
-
-message ObjectLocationStateUpdate {
+message ObjectLocationUpdate {
   bytes object_id = 1;
-  ObjectLocationState state = 2;
+  // When it's set, it indicates whether the object is added or removed from plasma store.
+  optional bool in_plasma = 2;
```

Comment:
Along the lines of what Sang suggested, I think it's clearer to keep using the `ObjectLocationState` and instead add a new type called `SPILLED`. The semantics for the `in_plasma` flag are a bit confusing.

Also, I'm realizing we already include the node ID in this request, so we could replace the previous `spilled_node_id` flag with a bool like `spilled_to_local_storage`.